eternalcore-api/src/main/java/com/eternalcode/core/feature/warp/WarpService.java

```diff
@@ -11,8 +11,14 @@ public interface WarpService {
 
     void removeWarp(String warp);
 
+    void addPermissions(String warp, String... permissions);
+
+    void removePermission(String warp, String permission);
+
     boolean warpExists(String name);
 
+    boolean doesWarpPermissionExist(String warp, String permission);
```

Comment:
Is permission needed here? We can just check it using isEmpty on permissions list in object